README.md

```diff
@@ -1,158 +1,552 @@
+
+## Who is it for?
+
+1. **as-code practitioners** - If you like *as-code philosophy* (infrastructure-as-code, pipelines-as-code) then you'll like Dojo, which provides **development and operations environment as code**.
+1. Consultants, DevOps Engineers, **anyone jumping projects and languages often**. With Dojo, getting the *right environment* for a project takes as long as the `docker pull`.
+1. **Polyglot software houses**, especially where CI agents are shared by a variety of projects.
+1. If you are already running builds with docker, then Dojo will make your scripts shorter and your life easier.
+
+By adopting Dojo you would be shifting responsibility for the development environment closer to the developers.
+A dojo docker image becomes a contract of what is a **correct environment** for a project **at a particular commit** (see [Dojofile](#dojofile)).
+
+# Table of contents
+
+1. [Quickstart](#quickstart)
+1. [Installation](#installation)
+1. [Dojofile](#dojofile)
+1. [Docker images](#docker-images)
+    * [Requirements](#image-requirements-and-best-practices)
+    * [Building images](#image-scripts) with `Dockerfile`
+        * Typical Dockerfile for [debian and ubuntu](#typical-debian-dockerfile)
+        * Typical Dockerfile for [alpine](#typical-alpine-dockerfile)
+1. [Secrets distribution](#secrets)
+1. [Drivers](#drivers)
+    * [docker](#docker-driver)
+    * [docker-compose](#docker-compose-driver)
+1. [Behavior reference](#behavior-reference)
+    * [CLI arguments](#cli-arguments)
+    * [Home and identity directory](#home-and-identity-directory)
+    * [Handling signals](#handling-signals)
+1. [FAQ](#faq)
+1. [Comparison to other tools](#comparison-to-other-tools)
+1. [Development](#Development)
+1. [License](#License)
+
+# Quickstart
+
+1. [Install docker](https://docs.docker.com/install/), if you haven't already.
+2. Install Dojo, it is a self-contained binary, so just place it somewhere on the `PATH`.
+On **Linux**:
+```bash
+DOJO_VERSION=0.4.0
+wget -O dojo https://github.com/ai-traders/dojo/releases/download/${DOJO_VERSION}/dojo_linux_amd64
+sudo mv dojo /usr/local/bin
+sudo chmod +x /usr/local/bin/dojo
+```
+
+Done. Now you have sufficient environment to build any project which leverages dojo images.
+
+Let's build this [java project](https://github.com/tomzo/gocd-yaml-config-plugin) using dojo:
+
+```bash
+git clone https://github.com/tomzo/gocd-yaml-config-plugin.git
+cd gocd-yaml-config-plugin
+dojo "gradle test jar"
+```
+
+The output should start with a docker command that was executed:
+```console
+tomzo@073c1c477b1f:/tmp/gocd-yaml-config-plugin$ dojo "gradle test jar"
+2019/04/28 14:01:40 [ 1]  INFO: (main.main) Dojo version 0.3.2
```

Comment:
Please change it to "Dojo version 0.4.0", because such a version was used above: "DOJO_VERSION=0.4.0". (You may want to run this experiment again with Dojo 0.4.0).